lib/avromatic/model/messaging.rb

```diff
@@ -0,0 +1,57 @@
+require 'avro_turf/messaging'
+
+module Avromatic
+  module Model
+
+    # This concern adds support for serialization based on AvroTurf::Messaging.
+    # This serialization leverages a schema registry to prefix encoded values
+    # with an id for the schema.
+    module Messaging
+      extend ActiveSupport::Concern
+
+      delegate :messaging, to: :Avromatic
+
+      module Encode
+        def avro_message_value
+          messaging.encode(
+            value_attributes_for_avro,
+            schema_name: value_avro_schema.fullname
+          )
+        end
+
+        def avro_message_key
+          raise 'Model has no key schema' unless key_avro_schema
+          messaging.encode(
+            key_attributes_for_avro,
+            schema_name: key_avro_schema.fullname
+          )
+        end
+      end
+      include Encode
+
+      # This module provides methods to message_decode an Avro-encoded value and
+      # an optional Avro-encoded key as a new model instance.
+      module Decode
+
+        # If two arguments are specified then the first is interpreted as the
+        # message key and the second is the message value. If there is only one
+        # arg then it is used as the message value.
+        def message_decode(*args)
```

Comment:
Should this be prefixed with `avro_` to minimize the likelihood of a collision with domain specific names?